api/src/queries/environments.js

```diff
@@ -1,75 +1,36 @@
-import { crudQueries } from 'co-postgres-queries';
+import client, { insertOne, updateOne } from "./knex";
 
-import db from './db';
-
-const query = crudQueries(
-    'environment',
-    ['name', 'state', 'project_id'],
-    ['id'],
-    ['id', 'name'],
-);
-
-query.selectPage = query.selectPage
-    .table('environment LEFT JOIN configuration ON (environment.id = configuration.environment_id)')
-    .idFields(['id'])
-    .searchableFields(['environment.name', 'project_id', 'environment.state'])
-    .returnFields([
-        'environment.id',
-        'environment.name',
-        `case
-            when count(configuration.name) = 0 then '[]'
-            else json_agg((
-                SELECT x
-                FROM (
-                    SELECT
-                        configuration.id,
-                        configuration.name,
-                        configuration.default_format as "defaultFormat"
-                    ) x
-                )) end as configurations`,
-    ])
-    .groupByFields(['environment.id', 'environment.name', 'environment.project_id']);
-
-const insertOne = async (environment) => {
-    const client = await db.link(query);
-    const result = await client.insertOne(environment);
-    client.release();
-
-    return result;
-};
-
-const updateOne = async (id, environment) => (await db.link(query)).updateOne(id, environment);
+const table = "environment";
+const fields = ["id", "name"];
 
 const findOne = async (projectId, environmentName) => {
-    const client = await db.link(query);
-    const result = await client.selectPage(undefined, undefined, {
-        project_id: projectId,
-        'environment.name': environmentName,
-        state: 'live',
+  const results = await client
+    .select(fields)
+    .from(table)
+    .where({
+      project_id: projectId,
+      state: "live",
+      name: environmentName
```

Comment:
'environment.name': environmentName, no need to join  ?

Reply:
Not in my knowledge